src/io/forsta/ccsm/database/DbBase.java

```diff
@@ -55,4 +55,16 @@ protected void removeAll(String table) {
     SQLiteDatabase db = mDbHelper.getWritableDatabase();
     db.delete(table, null, null);
   }
+
+  protected boolean doesRowExist(String table, String dbField, String fieldValue) {
+    SQLiteDatabase db = mDbHelper.getWritableDatabase();
+    String query = "Select * from " + table + " where " + dbField + " = " + fieldValue;
+    Cursor cursor = db.rawQuery(query,null);
+      if(cursor.getCount() <= 0) {
+        cursor.close();
+        return false;
+      }
+      cursor.close();
+      return true;
+  }
```

Comment:
I should make or find a method that returns the user ID when given the tag as a search criteria

Reply:
This is unnecessary. Just return the row from ContactDb if you find it, otherwise return null.